metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -14,42 +15,118 @@
 
 from metaflow.exception import MetaflowException
 from metaflow.metaflow_config import AWS_SANDBOX_ENABLED
+from metaflow.metaflow_config import BATCH_DESCRIBE_JOBS_THROTTLE_DELTA as DELTA
+from metaflow.metaflow_config import BATCH_DESCRIBE_JOBS_THROTTLE_TRIES as TRIES
+
+
+class Throttle(object):
+    def __init__(self, delta_in_secs=DELTA, num_tries=TRIES):
+        self._now = time.time()
+        self.delta_in_secs = int(delta_in_secs)  # Cast if from env variable
+        self.num_tries = int(num_tries)  # Cast if from env variable
```

Comment:
nit: i'd leave it to the caller to cast. For env vars, we usually do the casting in `metaflow_config.py`